server/src/main/java/com/paypal/selion/grid/servlets/GridStatistics.java

```diff
@@ -0,0 +1,152 @@
+import com.paypal.selion.pojos.BrowserStatisticsCollection;
+import com.paypal.selion.utils.ServletHelper;
+
+/**
+ * <code>GridStatistics</code> servlet displays the current load on the Grid per browser, i.e., the number of requests
+ * waiting on the queue for a browser and the maximum instances of that browser. The servlet responds only to client
+ * using accept header for all media types and accept header of type application/json. This servlet should be injected
+ * into the Grid. <br>
+ * 
+ * <pre>
+ * cURL clients
+ * 
+ * Sample requests
+ * curl -s http://<domain>:<port>/grid/admin/GridStatistics
+ * curl -s -X GET http://<domain>:<port>/grid/admin/GridStatistics
+ * curl -s -H "Accept: application/json" -X GET http://<domain>:<port>/grid/admin/GridStatistics
+ * 
+ * Browser clients
+ * 
+ * Go to the URL http://<domain>:<port>/grid/admin/GridStatistics
+ * 
+ * Sample response
+ * [{
+ *     "browserName": "chrome",
+ *     "statistics": {
+ *         "waitingRequests": 2,
+ *         "maxBrowserInstances": 10
+ *     }
+ * },
+ * {
+ *     "browserName": "firefox",
+ *     "statistics": {
+ *         "waitingRequests": 3,
+ *         "maxBrowserInstances": 15
+ *     }
+ * },
+ * {
+ *     "browserName": "internet explorer",
+ *     "statistics": {
+ *         "waitingRequests": 0,
+ *         "maxBrowserInstances": 1
+ *     }
+ * }]
+ * </pre>
+ */
+public class GridStatistics extends RegistryBasedServlet {
+
+    /**
+     * Serial Version ID
+     */
+    private static final long serialVersionUID = -4200130800419092658L;
+
+    public GridStatistics() {
+        this(null);
+    }
+
+    public GridStatistics(Registry registry) {
+        super(registry);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        process(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        process(req, resp);
+    }
+
+    /**
+     * This method gets all the nodes which are connected to the grid machine from the Registry and displays them in
+     * html page.
+     * 
+     * @param request
+     *            {@link HttpServletRequest} that represents the servlet request
+     * @param response
+     *            {@link HttpServletResponse} that represents the servlet response
+     * @throws IOException
+     */
+    protected void process(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String acceptHeader = request.getHeader("Accept");
+        if (acceptHeader != null && ((acceptHeader.contains("*/*")) || (acceptHeader.contains("application/json")))) {
+            ServletHelper.respondAsJsonWithHttpStatus(response, getGridLoadResponse(), HttpServletResponse.SC_OK);
+        } else {
+            response.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
```

Comment:
@mach6 We send only response in json format. So I concluded that acceptHeaders of type " */* " and " application/json " are the only valid client headers for our scenario.